src/Chapter4.hs

```diff
@@ -672,8 +694,24 @@ Specifically,
    subtree of a tree
  ❃ Implement the function to convert Tree to list
 -}
-
-
+data Tree a
+  = EmptyTree
+  | Tree a (Tree a) (Tree a)
+
+instance Functor Tree where
+  fmap :: (a -> b) -> Tree a -> Tree b
+  fmap _ EmptyTree = EmptyTree
+  fmap f (Tree a ll rl) = Tree (f a) (fmap f ll) (fmap f rl)
+
+reverseTree :: Tree a -> Tree a
+reverseTree EmptyTree = EmptyTree
+reverseTree (Tree n ll rl) = Tree n (reverseTree rl) (reverseTree ll)
+
+-- Not really sure about this one
+tree2list :: Tree a -> [a]
+tree2list EmptyTree = []
+tree2list (Tree a ll EmptyTree) = a : tree2list ll
+tree2list (Tree a EmptyTree rl) = a : tree2list rl
```

Comment:
Pattern matching in this function is not exhaustive, because you don't cover the case when both branches are not empty. you need to handle this case, and probably you can even see how some other cases can become redundant. But otherwise, looks good :+1:

Reply:
Yes! I had left that behind. Thank you!